src/V3Split.cpp

```diff
@@ -881,12 +945,73 @@ class SplitVisitor final : public SplitReorderBaseVisitor {
     // METHODS
 protected:
     virtual void makeRvalueEdges(SplitVarStdVertex* vstdp) override {
+        AstVarScope* const vscp = VN_CAST(vstdp->nodep(), VarScope);
+        UASSERT_OBJ(vscp, vstdp->nodep(), "RHS should be VarScope");
+        auto it = m_lhsRevision.find(vscp);
+        const int rev = it == m_lhsRevision.end() ? 0 : it->second;
         // Each 'if' depends on rvalues in its own conditional ONLY,
         // not rvalues in the if/else bodies.
         for (auto it = m_stmtStackps.cbegin(); it != m_stmtStackps.cend(); ++it) {
             const AstNodeIf* const ifNodep = VN_CAST((*it)->nodep(), NodeIf);
             if (ifNodep && (m_curIfConditional != ifNodep)) continue;
-            new SplitRVEdge(&m_graph, *it, vstdp);
+            new SplitRVEdge(&m_graph, *it, vstdp, rev);
+        }
+    }
+
+    void pruneDepsOnLastUpdatedSignalsInAlwaysComb() {
+        // If a variable is set only in the always_comb and all users of the variables refers the
+        // last assignment, prune the edge.
+        // e.g.
+        //   always_comb begin
+        //     sig_a = in_a + in_b;
+        //     sig_b = !sig_a;
+        //   end
+        // can be
+        //   always_comb sig_a = in_a + in_b;
+        //   always_comb sig_b = !sig_a;
+        // but
+        //   always_comb begin
+        //     sig_a = in_a + in_b;
+        //     sig_b = !sig_a;
+        //     sig_a = 0;
+        //   end
+        // cannot be split.
+        for (V3GraphVertex* vertexp = m_graph.verticesBeginp(); vertexp;
+             vertexp = vertexp->verticesNextp()) {
+            for (V3GraphEdge* edgep = vertexp->outBeginp(); edgep; edgep = edgep->outNextp()) {
+                SplitRVEdge* const rvEdgep = dynamic_cast<SplitRVEdge*>(edgep);
+                if (!rvEdgep) continue;
+                const SplitNodeVertex* const assignVertexp
+                    = dynamic_cast<const SplitNodeVertex*>(edgep->fromp());
+                UASSERT(assignVertexp, "must be SplitNodeVertex");
+                UASSERT(assignVertexp == vertexp, "must be same");
+                if (!VN_IS(assignVertexp->nodep(), Assign)) continue;
+                // fromp() is nondelayed (blocking) assignment
+                const SplitNodeVertex* const varScopeVertexp
+                    = dynamic_cast<const SplitNodeVertex*>(edgep->top());
+                UASSERT(varScopeVertexp, "top must be SplitNodeVertex");
```

Comment:
Please use UASSERT_OBJ instead where possible (in this case the vertex has a nodep() that can be used)

Reply:
I agree the rule and I am trying to keep it.
But this check is against the result of dynamic_cast from `V3GraphVertex *` to `SplitNodeVertex *`.
When the cast fails, `SplitNoeVertex::nodep()` is not available.